### PR TITLE
add unit tests coverage reports

### PR DESCRIPTION
**Pull request by yapei**
Hi @jhadvig added a new `coverage` script which can be used to run unit test and generate coverage report, I'm not sure if I should update existing `test` script or a new one. Could you help review?

### CODE REVIEW

**Comment by andrewballantyne** [comment]
From the description

> ...I'm not sure if I should update existing test script or a new one...

@yapei The `test.sh` in the root is what the CI bot triggers off of when a PR is created/updated. It's the `ci/prow/test` script after the various containers are created and setup stuff is done. If you would like the coverage to be part of work the developers do to the repo/PRs, we'll need it in that script.

I believe the files it generates are [.gitignore](https://github.com/openshift/hac-infra/blob/main/.gitignore#L15)'d though so they won't be committed. Hopefully whatever you're looking to do with them is at time of generation here in the hosted container we build/test in and nothing you need to be stored in git. 

I'm not quite sure what you're after, but I'd be happy to help if you can let me know what the process you're trying to apply is.

**Comment by jhadvig** [comment]
Per discussion with @yapei we should also incorporate the the coverage report upload. that would need to be part of the `test.sh`. Also linking the release PR thats updating ci-operator config for hac-infra.
https://github.com/openshift/release/pull/26709

**Comment by yapei** [comment]
**1. with the change of `collectCoverageFrom` now we have correct coverage report** credit to @andrewballantyne 
``` 
[fedora@preserved-ui-fedora35 hac-infra]$ yarn coverage
yarn run v1.22.17
$ jest --coverage --runInBand --detectOpenHandles --forceExit
 PASS  src/PluginEntry/PluginEntry.test.tsx (6.619 s)
  ✓ that the header presents the welcome message (57 ms)

------------------|---------|----------|---------|---------|-------------------
File              | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
------------------|---------|----------|---------|---------|-------------------
All files         |     100 |      100 |     100 |     100 |                   
 NotFound         |     100 |      100 |     100 |     100 |                   
  NotFound.tsx    |     100 |      100 |     100 |     100 |                   
  index.ts        |     100 |      100 |     100 |     100 |                   
 PluginEntry      |     100 |      100 |     100 |     100 |                   
  PluginEntry.tsx |     100 |      100 |     100 |     100 |                   
  index.ts        |     100 |      100 |     100 |     100 |                   
------------------|---------|----------|---------|---------|-------------------
Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        7.492 s
Ran all test suites.
Done in 8.67s.
```
**2. `yarn test` in `test.sh` is updated to `yarn coverage`**

it looks like we need have openshift/release#26709 to be merged first otherwise our `test` will fail with error
```
 ./test.sh: line 11: CODECOV_TOKEN: unbound variable 
```

File: test.sh [hunk 1]
```diff
@@ -3,5 +3,9 @@ set -euo pipefail
 
 yarn install
 yarn prod
-yarn test
+yarn coverage
 yarn lint
+
+curl -Os https://uploader.codecov.io/latest/linux/codecov
+chmod +x codecov
+./codecov -t ${CODECOV_TOKEN} --dir ./coverage
```
Comment (andrewballantyne) [inline test.sh:11]: Small nit, typically we want to end with a blank line so the next diff of anything added doesn't say it modified this line.

**Comment by andrewballantyne** [comment]
This looks good to me. Minor nit that could be completed or not. Just thinking forward to the next diff 🙂 
/lgtm

**Review by jhadvig — APPROVED** [review]
/lgtm
/approve

**Pull request merged**